pkg/apiserver/watch.go

```diff
@@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 				// client disconnect.
 				return
 			}
-			flusher.Flush()
+			if len(ch) == 0 {
```

Comment:
To your last question, yes, we merely want to ensure we always flush if we
receive the "last" event.  In all other cases Go will handle flushing for
us.

On Mon, May 23, 2016 at 1:14 PM, Xiang Li <notifications@github.com> wrote:

> In pkg/apiserver/watch.go
> <https://github.com/kubernetes/kubernetes/pull/26085#discussion_r64253208>
> :
>
> > @@ -196,7 +197,9 @@ func (s *WatchServer) ServeHTTP(w http.ResponseWriter, req *http.Request) {
> >  				// client disconnect.
> >  				return
> >  			}
> > -			flusher.Flush()
> > +			if len(ch) == 0 {
>
> also shouldnt http layer will do a auto flush if the http buffer is
> getting large?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/26085/files/c4bec1585fe9c31303c928c11ac45e380d2eaf77#r64253208>
>